Packs/OctoxLabs/Integrations/OctoxLabs/OctoxLabs.py

```diff
@@ -0,0 +1,223 @@
+    raise Exception("No command.")
+
+
+""" COMMAND FUNCTIONS """
+
+
+def test_module(octox: OctoxLabs, *_, **__) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type octox: ``octoxlabs.OctoxLabs``
+    :param octoxlabs.OctoxLabs: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+    octox.ping()
+    return "ok"
+
+
+def get_adapters(octox: OctoxLabs, *_, **__) -> CommandResults:
+    count, adapters = octox.get_adapters()
+
+    return CommandResults(
+        outputs_prefix="OctoxLabs.Adapters",
+        outputs={
+            "count": count,
+            "results": [
+                convert_to_json(
+                    obj=a,
+                    keys=[
+                        "id",
+                        "name",
+                        "slug",
+                        "description",
+                        "groups",
+                        "beta",
+                        "status",
+                        "hr_status",
+                    ],
+                )
+                for a in adapters
+            ],
+        },
+    )
+
+
+def get_connections(octox: OctoxLabs, args: Dict[str, Any]) -> CommandResults:
+    page = args.get("page", 1)
+    count, connections = octox.get_connections(page=page)
+
+    return CommandResults(
+        outputs_prefix="OctoxLabs.Connections",
+        outputs={
+            "count": count,
+            "results": [
+                convert_to_json(
+                    obj=c,
+                    keys=[
+                        "id",
+                        "adapter_id",
+                        "adapter_name",
+                        "name",
+                        "status",
+                        "description",
+                        "enabled",
+                    ],
+                )
+                for c in connections
+            ],
+        },
+    )
+
+
+def get_discoveries(octox: OctoxLabs, args: Dict[str, Any]) -> CommandResults:
+    page = args.get("page", 1)
+    count, discoveries = octox.get_discoveries(page=page)
+
+    return CommandResults(
+        outputs_prefix="OctoxLabs.Discoveries",
+        outputs={
+            "count": count,
+            "results": [
+                convert_to_json(
+                    d,
+                    keys=[
+                        "id",
+                        "start_time",
+                        "end_time",
+                        "status",
+                        "hr_status",
+                        "progress",
+                    ],
+                )
+                for d in discoveries
+            ],
+        },
+    )
+
+
+def get_last_discovery(octox: OctoxLabs, *_, **__) -> CommandResults:
+    discovery = octox.get_last_discovery()
```

Comment:
what happens if no results are found?

Reply:
Return error message as "Not Found."
Also see: [py-octoxlabs package](https://github.com/octoxlabs/py-octoxlabs/blob/main/src/octoxlabs/service.py#L51)